question15/main.js

```diff
@@ -0,0 +1,80 @@
+'use strict';
+
+const wrapper = document.getElementById('js-wrapper');
+const requestBtn = document.getElementById('js-request-btn');
+const requestNumber = document.getElementById('js-number');
+const requestText = document.getElementById('js-text');
+const modalOpen = document.getElementById('js-modal-btn');
+const mask = document.getElementById('js-mask');
+const modal = document.getElementById('js-modal');
+const modalClose = document.getElementById('js-close');
+
+  /* MODAL OPEN */
+modalOpen.addEventListener('click',()=>{
+    modal.classList.remove('hidden');
+    mask.classList.remove('hidden');
+});
+/* MODAL close */ 
+modalClose.addEventListener('click',()=>{
+    modal.classList.add('hidden');
+    mask.classList.add('hidden');
+});
+
+mask.addEventListener('click',()=>{
+    modalClose.click();
+});
+
+async function callApi() {
+    try {
+        const res = await fetch('https://myjson.dit.upm.es/api/bins/e8fj');
+        const json = await res.json();
+        return json.data;
+    }
+    catch(error) {
+        wrapper.textContent = "データが取得できませんでした";
+        console.error(error);
+    }
+    finally{
+        modalOpen.remove();
+    }
+}
+
+async function init(){
+    const result = await callApi();
+    if (result){
+        createList(result);
+    } else {
+        wrapper.textContent = "データの表示に失敗しました";
+        console.error('データの表示に失敗しました'); 
+    }
+}
+
+requestBtn.addEventListener('click',()=>{
+    if (! requestNumber.value === true || ! requestText.value === true){
+        alert('未入力箇所があります');
```

Comment:
With this check method, even if a space is entered in the name field, it will be sent.

By the way, as a side note, validation is possible by adding the required attribute to the input element. However, I think this time the specification is to check with JavaScript, so please change your JavaScript.

https://gray-code.com/html_css/specify-required-input-items-of-the-form/

Reply:
Thanks for the review! I totally had not noticed. I've added a validation method. Please confirm.
Latest commit: 90ef8ce